add keylime usage doc

DanNiESh opened the pull request.

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,97 @@
+Keylime
+=======
+
+ESI provides a guide to use `keylime`_ outside Ironic for run-time attestation with `IMA`_.
```
I think 'keylime' should be capitalized.

DanNiESh replied:
ack

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,97 @@
+-------------
+* A Keylime server machine with keylime registrar and verifier installed.
+* A node to be attested with an active TPM 2.0 and keylime agent installed.
+* A private network for attestation workflow.
```
It feels like the prerequisites should be the 'private attestation network', and then simply one node that will act as the Keylime server.

DanNiESh replied:
got it!

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,97 @@
+
+Build Keylime Server
+--------------------
+Install Keylime packages:
```
This section should probably start with provisioning the node that will act as the Keylime server; and then once provisioning is done, one ssh's into that server and does the rest here.

As an additional note, if that server needs to be connected to the public internet for this step, then a public network could be a requirement. Alternatively, the user might acquire keylime $some_other_way. You can describe that option and this one by noting that one can attach the node to the public network using 'esi node network attach'.

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,97 @@
+
+Alternatively, `Keylime ansible role`_ could be used to deploy Keylime against a hardware TPM.
+
+After Keylime agent is set up on the target node, check the keylime-agent log to get agent UUID.
```
What does it mean for the Keylime agent to be set up on the target node? I'd suggest detailing the commands to upload the image into glance; and then to provision the node-to-be-attested with the image.

tzumainn commented:
This looks pretty good! Some overall comments:

* make sure you detail ESI-specific commands and concerns; I listed some in my comments, but there could be more
* I think the structure could be different. Broadly speaking, I'd suggest:
   * Prerequisites (you already have this)
   * Setup (everything up to and including using DIB to create an image)
   * Usage (specifying that when you want to attest a node, you have to use metalsmith to provision using the specified image on the attestation network)

And then maybe one last step, which is mentioning that you can use 'esi node network attach' to attach the now-attested node to any other network.

DanNiESh commented:
> This looks pretty good! Some overall comments:
> 
> * make sure you detail ESI-specific commands and concerns; I listed some in my comments, but there could be more
> * I think the structure could be different. Broadly speaking, I'd suggest:
>   
>   * Prerequisites (you already have this)
>   * Setup (everything up to and including using DIB to create an image)
>   * Usage (specifying that when you want to attest a node, you have to use metalsmith to provision using the specified image on the attestation network)
> 
> And then maybe one last step, which is mentioning that you can use 'esi node network attach' to attach the now-attested node to any other network.

I've updated a new patch that addresses these comments.

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,135 @@
+
+.. prompt:: bash $
+
+  metalsmith deploy --image <image> --network <network> --resource-class baremetal --ssh-public-key <path_to_public_key>
```
I'd specify <attestation_network> here to make it clear where the node needs to end up.

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,135 @@
+-------------
+* A private attestation network.
+* A node specified as Keylime server.
+* The nodes-to-be-attested with an active TPM 2.0 and keylime agent installed.
```
Maybe an "(Optional) public network to acquire Keylime"?

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,135 @@
+
+  metalsmith deploy --image <image> --network <network> --resource-class baremetal --ssh-public-key <path_to_public_key>
+
+* Note: the server needs to be connected to the public internet in order to download Keylime packages. Attaching the node to the public network could be as simple as running the command ``openstack esi node network attach (--network <network> | --port <port>) <node>``.
```
This should go under "Install Keylime packages", since that's what this is for. It's a step before running git clone. Also, I'd just use the 'attach --network <public_network>' option, for simplicity.

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,135 @@
+
+  keylime_verifier
+  keylime_registrar
+
```
And here, specify that if the node is not on the attestation network, it should now be put there using 'openstack esi node network attach --network <attestation_network> <node>'

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,135 @@
+  keylime_verifier
+  keylime_registrar
+
+Setup Keylime Agent
```
Maybe 'Create Attestation Image', or some variation of that?

tzumainn commented on docs/source/keylime/index.rst:
```diff
@@ -0,0 +1,135 @@
+      --property ramdisk_id=$RAMDISK_ID \
+      keylime-image
+
+Alternatively, `Keylime ansible role`_ could be used to deploy Keylime against a hardware TPM.
```
Does this go under 'Build Keylime Server'? Has it been tested with a leased node?

DanNiESh replied:
This ansible role uses vagrant to create a VM with all keylime packages installed. The VM will be created inside the leased node. But that's not what people want. I guess I shouldn't mention this here.

tzumainn replied:
Yeah, if that's the case then I'd just drop it.

DanNiESh replied:
Maybe I could put this link here for manual installation? https://github.com/keylime/keylime#manual

tzumainn reviewed (changes requested):
Thanks for making the changes! A few more comments inline.

tzumainn commented:
Looks good - thanks!
